### PR TITLE
handle result type to HandleResult #48

### PR DESCRIPTION
**Pull request by venkattejaRaavi**
This pull request is about [**Change handle result type to HandleResult** #48 opened on Jul 29 by orkunkl.](https://github.com/CosmWasm/cosmwasm-template/issues/48). I have made some changes respective to the issue. Please give a feedback once the review is done.

### CODE REVIEW

**Comment by venkattejaRaavi** [comment]
@ethanfrey This pull request is about **[Change handle result type to HandleResult #48 opened on Jul 29 by orkunkl.](https://github.com/CosmWasm/cosmwasm-template/issues/48)** I have made some changes respective to the issue. Please give feedback once the review is done.

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
-name = "{{project-name}}"
+name = "cosmwasm-template"
```
Comment (ethanfrey) [inline Cargo.toml:2]: We need to leave these all as templates. You can run `cargo generate` and the generated file will have the proper values filled out.

File: hash.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+bc0254258140f04372d21a0a0e2b70db35931c1acacf00adc1d079722c5e528d  contract.wasm
```
Comment (ethanfrey) [inline hash.txt:1]: This and contract.wasm should not be committed.

File: src/contract.rs [hunk 3]
```diff
@@ -1,9 +1,5 @@
-use cosmwasm_std::{
-    to_binary, Api, Binary, Env, Extern, HandleResponse, InitResponse, MessageInfo, Querier,
-    StdResult, Storage,
-};
+use cosmwasm_std::{to_binary, Api, Binary, Env, Extern, HandleResponse, InitResponse, Querier, StdResult, Storage, MessageInfo, HandleResult, InitResult, StdError};
```
Comment (ethanfrey) [inline src/contract.rs:1]: It seems this is not proper with `cargo fmt`

File: src/contract.rs [hunk 4]
```diff
@@ -14,7 +10,7 @@ pub fn init<S: Storage, A: Api, Q: Querier>(
     _env: Env,
     info: MessageInfo,
     msg: InitMsg,
-) -> StdResult<InitResponse> {
+) -> InitResult {
```
Comment (ethanfrey) [inline src/contract.rs:13]: Please look at the issue again.

There was some discussion yesterday, and I think we prefer to move towards: `Result<InitResponse, StdError>` for consistency between all types.

File: src/contract.rs [hunk 5]
```diff
@@ -27,40 +23,38 @@ pub fn init<S: Storage, A: Api, Q: Querier>(
 // And declare a custom Error variant for the ones where you will want to make use of it
 pub fn handle<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
-    _env: Env,
+    env: Env,
     info: MessageInfo,
     msg: HandleMsg,
-) -> Result<HandleResponse, ContractError> {
+) -> HandleResult {
```
Comment (ethanfrey) [inline src/contract.rs:29]: This is no longer valid. We want to use `ContractError`. The previous version was correct

**Review by ethanfrey — COMMENTED** [review]
Thank you for your contribution.

This inspired us to revisit the issue and the requirements changed. This should be a quite minimal change of the return types and not so much code in any case.

Also, I should document the technique for updating the template. Basically, I generate a sample project (eg. "foo") and commit the initial commit. Then I make edits to the generated file and test with `cargo test` and fix with `cargo fmt`. When it looks like I want, then I use `git diff` to see the changes and manually apply them to the template. After doing this once, I commit the original template repo, and try generating it again, ensuring that `cargo build` and `cargo test` work out of the both, and `cargo fmt` and `cargo clippy` don't show any issues or make any changes.

**Comment by webmaster128** [comment]
Thank you for your PR @venkattejaRaavi. We need to get clarity on #48 first before we can continue here. In your PR your change the costom error types to StdError, which is not the direction we want to go. Each contract should define their own error types and test them in unit tests. If you see examples using `StdError`, those should be updated.

Closing this for now. Feel welcome to discuss with us in #48 what the best interface is.

**Pull request closed without merging**